### PR TITLE
reverting old current prices

### PR DESCRIPTION
**Pull request by josojo**
If our smart contract will be used as an oracle, this PR closes a back-door to introduce false prices.

Before this Pr, false prices could be broadcasted by submitting solutions with weird prices. Under certain circumstances, these prices would not be overwritten, even if the "malicious" solution is substituted by a better solution.

Testplan:
unit test

### CODE REVIEW

File: test/stablecoin_converter.js [hunk 1]
```diff
@@ -902,6 +902,52 @@ contract("StablecoinConverter", async (accounts) => {
 
     await stablecoinConverter.submitSolution(batchIndex, owner, orderId, volume, prices, tokenIdsForPrice)
   })
+  it("checks that currentPrices between different solutions are reset", async () => {
+    const feeToken = await MockContract.new()
+    const stablecoinConverter = await StablecoinConverter.new(2 ** 16 - 1, feeDenominator, feeToken.address)
```
Comment (fleupold) [inline test/stablecoin_converter.js]: If this is a constant, it should be caps?

File: test/stablecoin_converter.js [hunk 2]
```diff
@@ -902,6 +902,52 @@ contract("StablecoinConverter", async (accounts) => {
...
+    await stablecoinConverter.deposit(feeToken.address, 60000, { from: user_1 })
+    await stablecoinConverter.deposit(erc20_2.address, 60000, { from: user_2 })
+    await stablecoinConverter.deposit(erc20_3.address, 10000, { from: user_3 })
+
+    await stablecoinConverter.addToken(erc20_2.address)
+    await stablecoinConverter.addToken(erc20_3.address)
+
+    const batchIndex = (await stablecoinConverter.getCurrentStateIndex.call()).toNumber()
+
+    const orderId1 = await sendTxAndGetReturnValue(stablecoinConverter.placeOrder, 1, 0, true, batchIndex + 1, 59940, 60000, { from: user_1 })
```
Comment (fleupold) [inline test/stablecoin_converter.js]: Not for now, but in the long run, I think we should refactor our testdata to come from a common module where each test cases picks there scenario (e.g. 3 orders, 2 solutions) or whatever it needs.

**Review by fleupold — APPROVED** [review]
Nice!

<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia0.giphy.com%2Fmedia%2FlVBtp4SRW6rvDHf1b6%2Fgiphy-downsized-medium.gif&token=repair" data-gitmeme-token="repair"></a>

**Review by bh2smith — APPROVED** [review]
Looks good although I was not aware of this problem in the first place. Was this brought up as an issue before somewhere?

**Review by anxolin — COMMENTED** [review]
Also arrived late, good you saw this :)

**Pull request merged**